src/lib/Storage/DoctrineGateway.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzSupportTools\Storage;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\ParameterType;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\SPI\Persistence\Content\Type;
+
+class DoctrineGateway implements Gateway
+{
+    private const CONTENTOBJECT_TABLE = 'ezcontentobject';
+    private const USER_TABLE = 'ezuser';
+    private const CONTENT_TYPE_TABLE = 'ezcontentclass';
+    private const CONTENTOBJECT_VERSION_TABLE = 'ezcontentobject_version';
+
+    /** @var \Doctrine\DBAL\Connection */
+    private $connection;
+
+    /** @var \Doctrine\DBAL\Platforms\AbstractPlatform */
+    private $databasePlatform;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->databasePlatform = $connection->getDatabasePlatform();
+    }
+
+    public function getContentObjectsCount(bool $includeDrafts = false): int
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder
+            ->select($this->databasePlatform->getCountExpression('id'))
+            ->from(self::CONTENTOBJECT_TABLE);
+
+        if (!$includeDrafts) {
+            $queryBuilder->andWhere(
+                $queryBuilder->expr()->neq('status', ContentInfo::STATUS_DRAFT)
+            );
+        }
+
+        return (int) $queryBuilder->execute()->fetchColumn();
+    }
+
+    public function getUsersCount(): int
+    {
+        $queryBuilder = $this->connection->createQueryBuilder();
+        $queryBuilder
+            ->select($this->databasePlatform->getCountExpression('contentobject_id'))
+            ->from(self::USER_TABLE);
+
+        return (int) $queryBuilder->execute()->fetchColumn();
+    }
+
+    public function getDraftsCount(): int
```

Comment:
Hmm, good to have drafts, but might be even more interesting to know how many versions there are in the system in total as it is what will affects performance of SQL queries more.

Reply:
Added in https://github.com/ezsystems/ez-support-tools/pull/76/commits/2709ba354dd095baa067987521e3c67bef7e60e4.